microservices/payment-stripe/src/entities/coupon.ts

```diff
@@ -0,0 +1,116 @@
+import { IsTypeormDate, IsValidate } from '@lomray/microservice-helpers';
+import { IsEnum, IsInt, IsNumber, IsOptional, IsString, Length, Max, Min } from 'class-validator';
+import { JSONSchema } from 'class-validator-jsonschema';
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  JoinTable,
+  ManyToMany,
+  OneToMany,
+  PrimaryColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import CouponDuration from '@constants/coupon-duration';
+import Product from '@entities/product';
+import PromoCode from '@entities/promo-code';
+
+@JSONSchema({
+  properties: {
+    products: { $ref: '#/definitions/Product', type: 'array' },
+    promoCodes: { $ref: '#/definitions/PromoCode', type: 'array' },
+  },
+})
+@Entity()
+class Coupon {
+  @JSONSchema({
+    description: 'Field for storing id of according coupon entity created on payment service side',
+  })
+  @PrimaryColumn({ type: 'varchar', length: 8 })
+  @Length(1, 8)
+  couponId: string;
+
+  @Column({ type: 'varchar', nullable: true, default: null })
+  @IsString()
+  @IsOptional()
+  userId: string | null;
+
+  @JSONSchema({
+    description: 'Name of the coupon displayed to customers on for instance invoices or receipts',
+  })
+  @Column({ type: 'varchar', nullable: true })
```

Comment:
the same, @Nullable() decorator and varchar length?
```suggestion
  @Column({ type: 'varchar' })
```